File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, depth, value):
    if depth == len(numbers):
        return 1 if value == target else 0
    
    return (dfs(numbers, target, depth+1, value + numbers[depth])
          + dfs(numbers, target, depth+1, value - numbers[depth]))

def solution(numbers, target):
    return dfs(numbers, target, 0, 0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <bits/stdc++.h>

using namespace std;

int cal(vector<int> &numbers, int target, int depth, int val){
    if (depth == numbers.size())
        if (val == target) return 1;
        else return 0;
    
    return cal(numbers, target, depth+1, val+numbers[depth]) + 
        cal(numbers, target, depth+1, val-numbers[depth]);
}

int solution(vector<int> numbers, int target) {
    return cal(numbers, target, 0, 0);
}